### DISCUSSION TITLE
Console Color Configuration

### DISCUSSION THREAD

**Discussion by fossbrandon** [Tyrrrz/CliFx#126] [Q&A] [answered]
I'm wondering if anyone knows a way to configure the console colors at the `CliFx` layer before running any commands?

I started thinking about this after running into an issue where my CLI app briefly changes the foreground color before changing them back but it crashed before the reset leaving follow up runs of my app still using the changed color.

It would be nice to have the colors be configurable at the `CliFx` framework level so that no matter what the console colors currently are, they are reset when running the command again.

Does anyone have a way I could do this currently? If not, what do people think about adding this sort of configuration?

**Comment by Tyrrrz** [comment, marked as answer]
You can use these extension methods to set and auto-reset colors: https://github.com/Tyrrrz/CliFx/blob/33f95d941df4d76b4ec952617f5f38375e7d4204/CliFx/Infrastructure/IConsole.cs#L102-L137

Otherwise, a simple `try`/`finally` with `Console.ResetColors()` in `Program.cs` will suffice. Note that if your application hard-crashes (for example by getting the process killed from outside), then the `finally` blocks will not trigger regardless.
